gateleen-user/src/main/java/org/swisspush/gateleen/user/UserProfileHandler.java

```diff
@@ -186,8 +170,13 @@ protected void cleanupUserProfile(final JsonObject profile, final Handler<JsonOb
         JsonObject profileCopy = profile.copy();
         Set<String> profileFieldNames = profileCopy.fieldNames();
         profileFieldNames.stream().filter(fieldName -> !userProfileConfiguration.isAllowedProfileProperty(fieldName)).forEach(fieldName -> {
-            log.debug("Removing property '{}' from user profile", fieldName);
-            profile.remove(fieldName);
+            if(userProfileConfiguration.isRemoveNotAllowedProfileProperties()) {
+                log.debug("Removing property '{}' from user profile", fieldName);
+                profile.put(fieldName, null);
+            } else {
+                log.debug("Property '{}' is not allowed, but because of the `merge` option not removed from user profile", fieldName);
+                profile.remove(fieldName);
```

Comment:
Aren't those two statements contradicting? Log says we do NOT remove, but then code literally calls `remove`?
```java
    log ... not removed ... ;
    profile.remove(fieldName);
```